src/hooks/useSearch/reducer.ts

```diff
@@ -0,0 +1,55 @@
+import { filterRecords } from "./helpers";
+
+// -----------------------------------------------------------------------------
+
+type ReducerState = {
+  filteredRecords: any[];
+  searchTerm: string | null;
+  searchableFields: string[];
+  sourceRecords: any[];
+};
+
+export type SET_TERM = { type: "SET_TERM"; payload: string };
+export const SET_TERM = (payload: string): SET_TERM => {
+  return { type: "SET_TERM", payload };
+};
+
+export type CLEAR_TERM = { type: "CLEAR_TERM" };
+export const CLEAR_TERM = (): CLEAR_TERM => {
+  return { type: "CLEAR_TERM" };
+};
+
+/**
+ * A reducer function to handle search term actions in state management.
+ *
+ * @param {ReducerState} state - The current state, containing searchable fields, source records, and the current search term.
+ * @param {SET_TERM | CLEAR_TERM} action - The action to be processed. It can either set a new search term or clear the current search term.
+ * @returns {ReducerState} - The updated state based on the action type.
+ */
+export const reducer = (
```

Comment:
Depending on context, I find that a reducer can be a lot more productive for state tracking. There's a single source of truth for a component tree.